spring-beans/src/main/java/org/springframework/beans/BeanUtils.java

```diff
@@ -182,11 +183,18 @@
 		try {
 			ReflectionUtils.makeAccessible(ctor);
 			if (KotlinDetector.isKotlinReflectPresent() && KotlinDetector.isKotlinType(ctor.getDeclaringClass())) {
+				if (ObjectUtils.isEmpty(args)) {
+					return KotlinDelegate.instantiateClass(ctor);
```

Comment:
Given the fact that https://github.com/JetBrains/kotlin/pull/2855 has been closed because in practice it brings no benefit, is that part still relevant from your POV and if yes, could you elaborate on why and provide a JMH benchmark to quantify the gain?